packages/cli/src/lib/decorators/preconditions/apiKeyPrivilege.ts

```diff
@@ -0,0 +1,81 @@
+import PlatformClient, {
+  PrivilegeEvaluatorModel,
+} from '@coveord/platform-client';
+import Command from '@oclif/command';
+import {Config} from '../../config/config';
+import {AuthenticatedClient} from '../../platform/authenticatedClient';
+
+export const impersonatePrivilege = {
+  requestedPrivilege: {
+    owner: 'SEARCH_API',
+    targetDomain: 'IMPERSONATE',
+    targetId: '*',
+  },
+};
+
+export const createApiKeyPrivilege = {
+  requestedPrivilege: {
+    owner: 'PLATFORM',
+    targetDomain: 'API_KEY',
+    targetId: '*',
+    type: 'CREATE',
+  },
+};
+
+export function hasNecessaryCoveoPrivileges() {
+  return async function (target: Command) {
+    const authenticatedClient = new AuthenticatedClient();
+
+    const client = await authenticatedClient.getClient();
+    const {organization} = await getConfiguration(target);
+
+    if (!(await hasCreateApiKeyPrivilege(client, organization))) {
+      // TODO: better message
+      target.warn('You cannot create an API Key');
```

Comment:
Any idea of the message we could display?

Reply:
You could infer from the `anonymous` setting to see if the user has authenticated using an access-token or the API-Key.
Like this, you can do an actionnable to the user:
 - If he's OAuthed, tell him to go bother its Coveo Admin.
 - If he's API-Key-authed, prolly he created the API Key and derped, so tell him to modify it or submit another one